core/src/main/java/de/unistuttgart/iaas/amyassist/amy/core/pluginloader/PluginManager.java

```diff
@@ -0,0 +1,176 @@
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information see notice.md
+ */
+
+package de.unistuttgart.iaas.amyassist.amy.core.pluginloader;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+
+import javax.ws.rs.Path;
+
+import org.slf4j.Logger;
+
+import de.unistuttgart.iaas.amyassist.amy.core.IPlugin;
+import de.unistuttgart.iaas.amyassist.amy.core.configuration.ConfigurationLoader;
+import de.unistuttgart.iaas.amyassist.amy.core.di.DependencyInjection;
+import de.unistuttgart.iaas.amyassist.amy.core.di.annotation.PostConstruct;
+import de.unistuttgart.iaas.amyassist.amy.core.di.annotation.Reference;
+import de.unistuttgart.iaas.amyassist.amy.core.di.annotation.Service;
+import de.unistuttgart.iaas.amyassist.amy.core.plugin.api.SpeechCommand;
+import de.unistuttgart.iaas.amyassist.amy.core.speech.SpeechCommandHandler;
+import de.unistuttgart.iaas.amyassist.amy.httpserver.Server;
+
+/**
+ * Manages the plugin integration
+ * 
+ * @author Leon Kiefer
+ */
+@Service
+public class PluginManager {
+	private static final String CONFIG_NAME = "plugin.config";
+	private static final String CONFIG_PLUGINS = "plugins";
+	private static final String PROPERTY_PLUGINS_DIR = "pluginDir";
+
+	@Reference
+	private Logger logger;
+
+	@Reference
+	private PluginLoader pluginLoader;
+
+	@Reference
+	private Server server;
+	@Reference
+	private SpeechCommandHandler speechCommandHandler;
+	@Reference
+	private ConfigurationLoader configurationLoader;
+	@Reference
+	private DependencyInjection di;
+
+	private Properties config;
+
+	@PostConstruct
+	private void setup() {
+		this.config = this.configurationLoader.load(CONFIG_NAME);
+	}
+
+	/**
+	 * The project directory.
+	 */
+	public static final File projectDir = new File(".").getAbsoluteFile().getParentFile();
+
+	private List<String> getPluginListFromConfig() {
+		String[] plugins = this.config.getProperty(CONFIG_PLUGINS).split(",");
+		return Arrays.asList(plugins);
+	}
+
+	private boolean loaded = false;
+
+	/**
+	 * load the plugins
+	 */
+	public synchronized void loadPlugins() {
+		if (this.loaded) {
+			throw new IllegalStateException("the plugins are loaded");
+		}
+
+		this.logger.debug("projectDir: {}", projectDir);
+
+		if (!projectDir.exists()) {
+			this.logger.error("Project directory does not exist.");
+			return;
+		}
+
+		File pluginDir = new File(projectDir, this.config.getProperty(PROPERTY_PLUGINS_DIR));
+
+		if (!pluginDir.exists()) {
+			this.logger
+					.error("Plugin directory does not exist. Is the project path correct for your working directory?");
```

Comment:
Why formatted like this?

Reply:
this looks wrong here because the tab size on GitHub is 8